### PR TITLE
eigen: modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/eigen/all/conanfile.py [hunk 1]
```diff
@@ -10,19 +12,31 @@ class EigenConan(ConanFile):
...
     topics = ("eigen", "algebra", "linear-algebra", "vector", "numerical")
-    settings = "os", "compiler", "build_type", "arch"
-    options = {"MPL2_only": [True, False]}
-    default_options = {"MPL2_only": False}
-    exports_sources = ["patches/*"]
-    no_copy_source = True
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "MPL2_only": [True, False],
```
Comment (SpaceIm) [inline recipes/eigen/all/conanfile.py:18]: this option should not have been accepted in https://github.com/conan-io/conan-center-index/pull/4342, it's a pure interface definition without more logic like adding dependencies.

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]
LGTM

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/eigen/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/eigen/all/conanfile.py:18]: well, we have such options, which don't affect on package contents, but just add some additional preprocessor definitions to the package_info, it's fine

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/eigen/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/eigen/all/conanfile.py:18]: I think it makes recipes confusing and adds unnecessary complexity. Outside of conan, nobody would expect to see definition injected by another tool.

**Pull request merged**